### PR TITLE
test: Add unit tests for ts

### PR DESCRIPTION
**Pull request by Maledong**
Ref: https://github.com/eggjs/egg-cookies/pull/11.

After a long discussion with @whxaxes, I used a ts-node to dynamically
run and fetch the result to do comparations. This test mainly covers:

1) Intellisenses：
default

2) get/set methods.

### CODE REVIEW

File: test/lib/tsRun.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+describe('start running ts file under the ts folder...', () => {
+
+  it('should run successfully and generate a json file to compare values', () => {
+
+        // This will dynamically add cookieTest and have a run
+    require('ts-node').register({ typeCheck: true });
+    const result = require('../fixtures/ts/cookiesTest');
+
+        // Check one by one
```
Comment (whxaxes) [inline test/lib/tsRun.test.js]: 一些缩进对齐一下，其他 LGTM

**Comment by whxaxes** [comment]
tsRun.test.js 改名为 ts.test.js 呗

**Comment by atian25** [comment]
cool

**Comment by atian25** [comment]
@Maledong thank you for the patience you showed when we were working on this problem.

**Comment by Maledong** [comment]
No problem, I don't wanna leave it alone. And I'll fix that.

File: test/lib/tsRun.test.js (hunk 1 above)
Comment (Maledong) [reply test/lib/tsRun.test.js]: OK

**Comment by Maledong** [comment]
@atian25 : No problem. After merge this, please @me and I'll upgrade the egg version by fetching the latest egg-cookie.

**Comment by atian25** [comment]
land at 2.2.3

**Comment by Maledong** [comment]
@atian25：Just want to confirm have you really released this version? After I upgrade to the latest version of egg, I found that 'egg-cookies' has no 'index.d.ts'？

**Comment by atian25** [comment]
e... last PR is missing `files`...

**Comment by Maledong** [comment]
No problem, a haha …… Thanks anyway!

**Comment by atian25** [comment]
https://github.com/eggjs/egg-cookies/pull/14

**Comment by Maledong** [comment]
@atian25：Are you publishing it to npm? Just I saw another 'egg-cookie' 0.0.1. Maybe this can be disclosed :)

**Comment by atian25** [comment]
image

**Comment by atian25** [comment]
2.2.4

**Comment by Maledong** [comment]
Yes, we're now using egg-cookies instead of egg-cookie. So this is useless in order to be mistaken. You can declare that's deprecated, just a notify.

**Comment by Maledong** [comment]
Thanks! I'll see and fix the test :)

**Comment by atian25** [comment]
```bash
➜  npm deprecate egg-cookie 'use egg-cookies instead'

➜  npm info egg-cookie deprecated
use egg-cookies instead
```

File: package.json [hunk 2]
```diff
@@ -30,16 +33,16 @@
...
     "ci": "npm run lint && npm run cov",
     "autod": "autod"
   },
   "engines": {
-    "node": ">= 4.0.0"
+    "node": ">= 8.0.0"
   },
   "ci": {
-    "version": "4, 6, 7"
+    "version": "8, 9"
```
Comment (atian25) [inline package.json]: 这个地方有问题。。。 这是 break change 了

Comment (atian25) [reply package.json]: https://github.com/eggjs/egg-cookies/pull/16

**Pull request merged**